File: jdroid-gradle-commons-plugin/src/main/groovy/com/jdroid/gradle/commons/tasks/BuildScriptDependenciesTask.java
package com.jdroid.gradle.commons.tasks;

import org.codehaus.groovy.runtime.DefaultGroovyMethods;
import org.gradle.api.artifacts.Configuration;

public class BuildScriptDependenciesTask extends AbstractTask {

	@Override
	protected void onExecute() {
		Configuration classpathConfiguration = getProject().getBuildscript().getConfigurations().findByName("classpath");
		if (classpathConfiguration != null) {
			for (String each : classpathConfiguration.getAsPath().split(":")) {
				DefaultGroovyMethods.println(this, each);
			}
		}
	}

}

File: jdroid-gradle-commons-plugin/src/main/groovy/com/jdroid/gradle/commons/tasks/BuildScriptDependenciesTask.kt
package com.jdroid.gradle.commons.tasks

open class BuildScriptDependenciesTask : AbstractTask() {

    override fun onExecute() {
        val classpathConfiguration = project.buildscript.configurations.findByName("classpath")
        if (classpathConfiguration != null) {
            for (each in classpathConfiguration.asPath.split(":")) {
                println(each)
            }
        }
    }
}
